processing/src/main/java/org/apache/druid/query/filter/EqualityFilter.java

```diff
@@ -0,0 +1,578 @@
+  {
+    DimFilter.DimFilterToStringBuilder bob = new DimFilter.DimFilterToStringBuilder().appendDimension(
+                                                                                         column,
+                                                                                         extractionFn
+                                                                                     )
+                                                                                     .append(" = ")
+                                                                                     .append(matchValue);
+
+    if (!ColumnType.STRING.equals(matchValueType)) {
+      bob.append(" (" + matchValueType.asTypeString() + ")");
+    }
+    return bob.appendFilterTuning(filterTuning).build();
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    EqualityFilter that = (EqualityFilter) o;
+    if (!column.equals(that.column)) {
+      return false;
+    }
+    if (!Objects.equals(matchValueType, that.matchValueType)) {
+      return false;
+    }
+    if (!Objects.equals(extractionFn, that.extractionFn)) {
+      return false;
+    }
+    if (!Objects.equals(filterTuning, that.filterTuning)) {
+      return false;
+    }
+    if (matchValueType.isArray()) {
+      // just use predicate to see if the values are the same
+      final ExprEval<?> thatValue = ExprEval.ofType(
+          ExpressionType.fromColumnType(that.matchValueType),
+          that.matchValue
+      );
+      final Predicate<Object[]> arrayPredicate = predicateFactory.makeArrayPredicate(matchValueType);
+      return arrayPredicate.apply(thatValue.asArray());
+    } else {
+      return Objects.equals(matchValue, that.matchValue);
+    }
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(column, matchValueType, matchValue, extractionFn, filterTuning);
+  }
+
+  @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    if (!Objects.equals(getColumn(), dimension) || getExtractionFn() != null) {
+      return null;
+    }
+    RangeSet<String> retSet = TreeRangeSet.create();
+    retSet.add(Range.singleton(String.valueOf(matchValue)));
+    return retSet;
+  }
+
+  @Nullable
+  @Override
+  public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
+  {
+    if (!Filters.checkFilterTuningUseIndex(column, selector, filterTuning)) {
+      return null;
+    }
+
+    final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(column);
+    if (indexSupplier == null) {
+      return Filters.makeNullIndex(false, selector);
+    }
+
+    final StringValueSetIndex valueSetIndex = indexSupplier.as(StringValueSetIndex.class);
+    if (valueSetIndex == null) {
+      // column exists, but has no index
+      return null;
+    }
+    return valueSetIndex.forValue(String.valueOf(matchValue));
```

Comment:
I had expected that this would create an `ObjectValueSetIndex` or something that would enable us to just pass in the object and get indexes.  Why not do that?

Reply:
100% my plan, i just haven't finished implementing it yet

Reply:
ive added `TypedValueIndex` that pushes down the native match type and match value type so that the column can handle comparison as appropriate